### PR TITLE
Update Rails, Webpacker and ReScript

### PR DESCRIPTION
**Pull request by gscarv13**
#### The goal of this PR is to update:
- Rails from 5 to 6 
- Webpacker
- Webpack configuration
- Replace Reason with ReScript

### CODE REVIEW

**Comment by alex35mil** [comment]
@gscarv13 Is it possible to submit a separate PR for ReScript related changes? I'm not the best person to review the Ruby/Rails parts.

**Review by justin808 — APPROVED** [review]
LGTM

**Comment by justin808** [comment]
Hi @alexfedoseev, The Rails parts are boilerplate updates. I approved those in 3 min. 

Please skip the ruby files and do a cursory review.

File: app/javascript/bundles/HelloWorld/HelloWorld.res [hunk 1]
```diff
@@ -0,0 +1,14 @@
+@react.component
+let make = (~nameProp: string) => {
+  let (name, setName) = React.useState(() => nameProp)
+
+  <div>
+    <h3> {("Hello, " ++ name ++ "!")->React.string} </h3>
```
Comment (alex35mil) [inline app/javascript/bundles/HelloWorld/HelloWorld.res:6]: It's better to use interpolation:

```rescript
`Hello, ${name}!`
```

**Review by alex35mil — COMMENTED** [review]
lgtm

**Pull request merged**